src/Missings.jl

```diff
@@ -208,6 +208,91 @@ missing
 """
 passmissing(f) = PassMissing{Core.Typeof(f)}(f)
 
+struct SpreadMissings{F} <: Function
+    f::F
+end
+
+function (f::SpreadMissings{F})(xs...; kwargs...) where {F}
+    if any(x -> x isa AbstractVector{>:Missing}, xs)
+        vecs = Base.filter(x -> x isa AbstractVector, xs)
```

Comment:
Yeah. But indeed that would be quite restrictive. Maybe we can hardcode a list of allowed types for now, or conversely throw an error for some collection types like `Tuple`, etc.? It's not clear to me yet whether it would be more useful to treat them like scalars or like vectors.

The list of hardcoded behaviors for `broadcasted` is here:
https://github.com/JuliaLang/julia/blob/8197c418e1d806f5faf1fd900c78068ea8046e1d/base/broadcast.jl#L700-L705

Reply:
Okay let me know if you think the updated `non_spreadable_check` is okay.